torch/_refs/__init__.py

```diff
@@ -3551,18 +3561,20 @@ def repeat(a: Tensor, *repeat_shape) -> Tensor:
 
 
 def _reshape_view_helper(a: TensorLikeType, *shape, allow_copy: bool) -> TensorLikeType:
+    from torch.fx.experimental.symbolic_shapes import guard_size_oblivious, sym_eq
+
     # Creates a valid shape
     shape = utils.extract_shape_from_varargs(shape, validate=False)
     # Reshape may be given a shape with a -1 length
     # This indicates that the dimension's length should be inferred
     shape = utils.infer_size(shape, a.numel())
 
     # Short-circuits if shape is the same
-    if tuple(a.shape) == tuple(shape):
+    if guard_size_oblivious(sym_eq(tuple(a.shape), tuple(shape))):
```

Comment:
Why `sym_eq` here rather than `==`?

Reply:
Equality on a list will result in `and` ing together the intermediate booleans together, which will cause them to get guarded on. I need to delay the guard until it gets to the guard_size_oblivious call.